### ISSUE TITLE
Loading spinner does not show when photo source is set back to nil after being set to some value

### ISSUE THREAD

**Issue by brianneisler** [NYTimes/NYTPhotoViewer#186]
Steps to reproduce
- init photosViewController with a photo that has `photo.image = nil;`
- present PhotoViewController
- update photo.image 
```objective-c    
photo.image = [UIImage imageWithData: data];
[self.photoViewer updateImageForPhoto:photo];
```
- dismiss PhotoViewController
- set photo.image back to nil
```objective-c  
photo.image = nil; 
[self.photoViewer updateImageForPhoto:photo];
```
- present PhotoViewController
- update photo.image to some new source
```objective-c    
photo.image = [UIImage imageWithData: data];
[self.photoViewer updateImageForPhoto:photo];
```

What should happen
- the spinner shows up the first time
- the spinner should also show up the second time when the photo source is set back to nil

What actually happens
- the spinner shows up the first time
- the photo viewer is black for a moment and then the new image shows up

**Comment by cdzombak** [comment]
I'm 👍 on this. @dzlobin / @soundsokay, any reservations?

**Comment by cdzombak** [comment]
closed by #187; thanks for your contribution, @brianneisler!
